### PR TITLE
Boringssl merge 7ada8466927864b6427277f455483c61081a32db

### PR DESCRIPTION
**Pull request by samuel40791765**
*Description of changes:*
This merges in the complete history up to 2021-6-8 `7ada8466927864b6427277f455483c61081a32db`.

Ran the following:
```
git fetch git@github.com:google/boringssl.git ${COMMIT_HASH} 
git merge --strategy recursive --allow-unrelated-histories ${COMMIT_HASH}
python3 ./util/generate_build_files.py
git commit -m "Update build files in generated-src"
git push --set-upstream origin "boringssl-merge-${COMMIT_HASH}"
```

- Ran the following [runbook](https://github.com/awslabs/aws-lc/blob/main/util/fipstools/delocate/README.md) to regenerate `delocate.peg.go`. 
- This includes https://github.com/google/boringssl/commit/597ffef971dd980b7de5e97a0c9b7ca26eec94bc commit which fixes https://github.com/awslabs/aws-lc/issues/172.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dkostic — COMMENTED** [review]
I did the first pass, will continue tomorrow.

File: crypto/CMakeLists.txt [hunk 1]
```diff
@@ -75,19 +75,19 @@ if(PERL_EXECUTABLE)
     endif()
```
[the remark this answers is not in the document]
Comment (samuel40791765) [reply crypto/CMakeLists.txt:75]: The indentations were something accidentally leaked in there while I was resolving a merge conflict. The only new line is `conf/conf_test.cc` which BoringSSL added.
I've updated the file and will push when Torben's done reviewing.

File: include/openssl/base.h [hunk 2]
```diff
@@ -202,7 +198,7 @@ extern "C" {
 // against multiple revisions of BoringSSL at the same time. It is not
 // recommended to do so for longer than is necessary.
 
-#define AWSLC_API_VERSION 14
+#define AWSLC_API_VERSION 16
```
[the remark this answers is not in the document]
Comment (samuel40791765) [reply include/openssl/base.h:201]: Yes, we updated this to align with BoringSSL's `BORINGSSL_API_VERSION` which has been updated to 16.

File: crypto/err/ssl.errordata [hunk 3]
```diff
@@ -59,6 +59,7 @@ SSL,264,DUPLICATE_KEY_SHARE
 SSL,296,DUPLICATE_SIGNATURE_ALGORITHM
 SSL,283,EARLY_DATA_NOT_IN_USE
 SSL,144,ECC_CERT_NOT_FOR_SIGNING
+SSL,319,ECH_REJECTED
```
[the remark this answers is not in the document]
Comment (samuel40791765) [reply crypto/err/ssl.errordata:62]: Sure, do you mean to add this as an enforcing function every time we merge? Or is there anything we want to remove here in this merge

File: crypto/fipsmodule/rand/internal.h [hunk 4]
```diff
@@ -45,12 +45,10 @@ void RAND_bytes_with_additional_data(uint8_t *out, size_t out_len,
 // for seeding a DRBG, to |out_entropy|. It sets |*out_used_cpu| to one if the
 // entropy came directly from the CPU and zero if it came from the OS. It
 // actively obtains entropy from the CPU/OS and so should not be called from
-// within the FIPS module if |BORINGSSL_FIPS_PASSIVE_ENTROPY| is defined.
+// within the FIPS module.
 void CRYPTO_get_seed_entropy(uint8_t *out_entropy, size_t out_entropy_len,
                              int *out_used_cpu);
 
-#if defined(BORINGSSL_FIPS_PASSIVE_ENTROPY)
```
[the remark this answers is not in the document]
Comment (torben-hansen) [reply crypto/fipsmodule/rand/internal.h:52]: It does but I expected that. Will have to revisited the branch and update it.

File: crypto/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (torben-hansen) [reply crypto/CMakeLists.txt:75]: FYI: can't see you are obs on the indentation if the conversation is marked as resolved :)

File: crypto/err/ssl.errordata (hunk 3 above)
[the remark this answers is not in the document]
Comment (torben-hansen) [reply crypto/err/ssl.errordata:62]: I dunno how you would do that programatically. There are lot's of values defined as macros, that really defines a namespace and should be distinct.

File: crypto/fipsmodule/CMakeLists.txt [hunk 5]
```diff
@@ -205,12 +205,15 @@ if(FIPS_DELOCATE)
...
+  # If building with OPENSSL_NO_ASM then ARCH will be "generic", but we still
+  # need to use SHA-256. Since this only matters for FIPS, we only need to
+  # worry about the Linux spelling of AArch64.
+  if (ARCH STREQUAL "aarch64" OR ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "aarch64")
     set(MAYBE_INJECT_HASH_SHA256_FLAG "-sha256")
   endif()
 
   go_executable(inject_hash
-	        boringssl.googlesource.com/boringssl/util/fipstools/inject_hash)
+                boringssl.googlesource.com/boringssl/util/fipstools/inject_hash)
```
Comment (torben-hansen) [inline crypto/fipsmodule/CMakeLists.txt:216]: indentation?

File: crypto/rand_extra/passive.c [hunk 6]
```diff
@@ -15,7 +15,7 @@
 #include <openssl/base.h>
 #include "../fipsmodule/rand/internal.h"
 
-#if defined(BORINGSSL_FIPS_PASSIVE_ENTROPY)
+#if defined(BORINGSSL_FIPS)
```
[the remark this answers is not in the document]
Comment (torben-hansen) [reply crypto/rand_extra/passive.c:18]: Let's do it

File: go.mod [hunk 7]
```diff
@@ -2,4 +2,7 @@ module boringssl.googlesource.com/boringssl
 
 go 1.13
 
-require golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
+require (
+	golang.org/x/crypto v0.0.0-20210513164829-c07d793c2f9a
+	golang.org/x/net v0.0.0-20210614182718-04defd469f4e
```
Comment (torben-hansen) [inline go.mod:7]: Hope this works int.

Comment (torben-hansen) [reply go.mod:7]: Further thoughts: this would mean changes to int build.

**Review by torben-hansen — COMMENTED** [review]
lgtm, but should fix indentation.

File: crypto/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (samuel40791765) [reply crypto/CMakeLists.txt:75]: Oh sorry, that wasn't intentional 😅

File: crypto/fipsmodule/CMakeLists.txt (hunk 5 above)
Comment (samuel40791765) [reply crypto/fipsmodule/CMakeLists.txt:216]: I think this one's an intentional one from BoringSSL: https://github.com/google/boringssl/blob/master/crypto/fipsmodule/CMakeLists.txt#L210

**Review by torben-hansen — APPROVED** [review]

**Review by dkostic — APPROVED** [review]

**Pull request merged**